src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -102,7 +102,7 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
         # create the RGW keyring
         rgw_user = f'{rados_user}-rgw'
         rgw_keyring = self.create_rgw_keyring(daemon_spec)
-        if spec.virtual_ip:
+        if spec.virtual_ip and not spec.enable_haproxy_protocol:
```

Comment:
I understand why we're adding the vip to the haproxy hosts list further down, but I don't understand why we need to not set the VIP from the spec as the bind address for ganesha in haproxy protocol mode.